server/src/main/java/org/elasticsearch/index/mapper/TimeSeriesIdFieldMapper.java

```diff
@@ -166,19 +167,22 @@ protected String contentType() {
         return CONTENT_TYPE;
     }
 
-    public static void encodeTsid(StreamOutput out, SortedMap<String, BytesReference> dimensionFields) throws IOException {
+    public static void encodeTsid(StreamOutput out, SortedMap<BytesRef, BytesReference> dimensionFields) throws IOException {
         out.writeVInt(dimensionFields.size());
-        for (Map.Entry<String, BytesReference> entry : dimensionFields.entrySet()) {
-            String fieldName = entry.getKey();
-            BytesRef fieldNameBytes = new BytesRef(fieldName);
-            int len = fieldNameBytes.length;
-            if (len > DIMENSION_NAME_LIMIT) {
+        for (Map.Entry<BytesRef, BytesReference> entry : dimensionFields.entrySet()) {
+            BytesRef fieldName = entry.getKey();
+            if (fieldName.length > DIMENSION_NAME_LIMIT) {
                 throw new IllegalArgumentException(
-                    "Dimension name must be less than [" + DIMENSION_NAME_LIMIT + "] bytes but [" + fieldName + "] was [" + len + "]."
+                    String.format(
+                        Locale.ROOT,
+                        "Dimension name must be less than [%s] bytes but [%s] was [%s].",
+                        DIMENSION_NAME_LIMIT,
+                        fieldName.utf8ToString(),
+                        fieldName.length
```

Comment:
The formatter made a mess of this when I made the parameters a little longer so I switched it to String.format so human can read the message.